.proposals/records/index-scaling-v1.md

```diff
@@ -0,0 +1,235 @@
+# Index Scaling: Decoupling disk, reads/writes 
+
+This proposal focuses on decoupling shard disk storage from the IndexNodes
+they are being written to and also enables scaling reads separately from writes.
+
+It primarily accomplishes this by using a replicated shared write storage layer and allowing
+read replicas of the file storage layer to enable decoupled ownership of shard data. This enables\
+scaling index read replicas separately from writes.
+
+
+*Goals*:
+- Allow short term scaling needs of IndexNode and search requests
+- Allow to scale reads separately from writes without rebalancing churn
+- Minimize impact of strategy on existing design to validate strategy without too much refactoring churn
+
+
+*Assumptions*:
+- Reads/writes should be scaled separately
+- (We will read much more than we will write). Is this true?
```

Comment:
We should have metrics in place for this, but what I was assuming so far:

 - In general, a particular document will be written once and read (searched) many times
 - Another thing is the global write rates Vs search rates that the IndexNode may receive: They may be for sure different and it is a reason already to aim to scale both separately.